UNITTESTS/fakes/events/EventQueue.cpp

```diff
@@ -0,0 +1,114 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "EventQueue.h"
+
+namespace events {
+
+handle_t EventQueue::call_handler(function_t handler)
+{
+    return call_handler_in(0, handler);
+}
+
+handle_t EventQueue::call_handler_in(tick_t ms, function_t handler)
+{
+    _handler_id++;
+
+    _handlers.push_back(
+        internal_event{
+            std::unique_ptr<function_t>(new function_t(handler)),
+            _now + ms,
+            _handler_id
+        }
+    );
+
+    return _handler_id;
+}
+
+bool EventQueue::cancel_handler(handle_t handle)
+{
+    if (!handle) {
+        return false;
+    }
+
+    _handlers.erase(
+        std::remove_if(
```

Comment:
Can we split the two operations ? If `remove_if` return `_handlers.end()` then false should be returned.

Reply:
I return false if failed to find